### PR TITLE
🚀 🚀 🚀 🚀  <template name=value> render improvement!!!!

### PR DESCRIPTION
**Pull request by snuggs**
## BEFORE 1,000 Records _800ms_
capture d ecran 2017-07-19 a 21 46 56

## AFTER 1,000 Records _(56-100ms)_

capture d ecran 2017-07-19 a 18 34 30
capture d ecran 2017-07-19 a 21 01 22
capture d ecran 2017-07-19 a 18 36 01

### CODE REVIEW

File: examples/infinity-calendar/infinity-calendar.es [hunk 1]
```diff
@@ -64,12 +64,20 @@ Element `infinity-calendar`
     const
       next_month = this.month + 1
     , date = new Date (this.year , next_month, 0)
-    , length = { length: date.getDate () }
+  //, length = { length: date.getDate () }
```
Comment (snuggs) [inline examples/infinity-calendar/infinity-calendar.es]: OH YEAH We DEFINITELY don't want to commit THIS file @brandondees @tmornini

File: elements/html-template-element.es [hunk 2]
```diff
@@ -1,51 +1,51 @@
...
+    , contexts = [].concat ( ... [context] )   // https://dom.spec.whatwg.org/#converting-nodes-into-a-node
+    , keys     = Object.keys (contexts [0])    // memoize keys
+    , tokens   = keys.map (key => '{'+key+'}') // memoize tokens
+    , fragment = document.createElement ('template')
 
-    template.innerHTML =
-    contexts
-      .map (context => context)
-      .map ((context, index) => {
+    , deposit = (context, index,) => {
```
Comment (snuggs) [inline elements/html-template-element.es]: trailing comma. amazed this works. i think is the secret behind `...params` "rest params" https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters

**Comment by snuggs** [comment]
I mean if we WERE comparing...which we are not.. ;-) 

image

File: examples/infinity-calendar/infinity-calendar.es [hunk 3]
```diff
@@ -64,12 +64,20 @@ Element `infinity-calendar`
...
-      .apply (null, length)
-      .map (days)
+      const d = []
+
+      for (var i = 0; i < 10; i++)
+        d.push (days ('', i))
+
+      console.log (d)
+      return d
+//  return Array
```
Comment (pachonk) [inline examples/infinity-calendar/infinity-calendar.es]: Should this commented code be here?

**Comment by snuggs** [comment]
@pachonk the whole thing shouldn't be there. Nice catch on that file

File: elements/html-template-element.es [hunk 4]
```diff
@@ -1,51 +1,58 @@
 // INTERESTING! Converting `Template` to a class increases size by ~16 octets
 
-const HTMLTemplateElement = Template = function (name) {
+const Template = HTMLTemplateElement = function (name) {
 
   // create shallow clone using `.getOwnPropertyDescriptors`
   // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors#Examples
   // https://docs.microsoft.com/en-us/scripting/javascript/reference/object-getownpropertydescriptor-function-javascript
+  // NO IE SUPPORT!!!!
```
Comment (pachonk) [inline elements/html-template-element.es]: Should only have 3 `!`

**Pull request merged**